### PR TITLE
money_trace: allow custom header, change span-id gen, opt to create if none

### PR DESCRIPTION
**Pull request by traeak**
This enhances the money_trace to allow it to run globally, with the ability to override the global with a remap plugin.

Also adds in better logging support for the current transaction's trace header
and adds a transparent mode with --passthru.

New options:

- --create-if-none=[true|false] (default: false)
- --global-skip-header=[name] (prefer internal '@' header)
- --header=[<name>] (default: X-MoneyTrace)
- --passthru[true|false] (default: false)
- --pregen-header=[name] (prefer internal '@' header, for example: %<{@pregen}cqh>

Follows the syntax outlined here: https://github.com/Comcast/money except the current state machine id or (crid log field) is used instead of a random number for the span-id.

### CODE REVIEW

**Comment by traeak** [comment]
[approve ci]

**Comment by traeak** [comment]
[approve ci ubuntu]

**Comment by bneradt** [comment]
[approve ci]

File: tests/gold_tests/pluginTest/money_trace/money_trace.test.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+#  limitations under the License.
+import os
+Test.Summary = '''
+Test money_trace remap
+'''
+
+# Test description:
+
+Test.SkipUnless(
+    #    Condition.PluginExists('xdebug.so'),
```
Comment (randall) [inline tests/gold_tests/pluginTest/money_trace/money_trace.test.py:24]: remove commented out code?

File: tests/gold_tests/pluginTest/money_trace/money_trace.test.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    f"map http://header/ http://127.0.0.1:{server.Variables.Port} @plugin=money_trace.so @pparam=--header=mt",
+    f"map http://pregen/ http://127.0.0.1:{server.Variables.Port} @plugin=money_trace.so @pparam=--pregen-header=@pregen",
+    f"map http://pgh/ http://127.0.0.1:{server.Variables.Port} @plugin=money_trace.so @pparam=--header=mt @pparam=--pregen-header=@pregen",
+    f"map http://create/ http://127.0.0.1:{server.Variables.Port} @plugin=money_trace.so @pparam=--create-if-none=true",
+    f"map http://cheader/ http://127.0.0.1:{server.Variables.Port} @plugin=money_trace.so @pparam=--create-if-none=true @pparam=--header=mt",
+    f"map http://cpregen/ http://127.0.0.1:{server.Variables.Port} @plugin=money_trace.so @pparam=--create-if-none=true @pparam=--pregen-header=@pregen",
+    f"map http://passthru/ http://127.0.0.1:{server.Variables.Port} @plugin=money_trace.so @pparam=--passthru=true",
+])
+
+# ts.Disk.plugin_config.AddLine("xdebug.so")
```
Comment (randall) [inline tests/gold_tests/pluginTest/money_trace/money_trace.test.py:72]: remove commented out code?

File: tests/gold_tests/pluginTest/money_trace/money_trace.test.py (hunk 1 above)
Comment (traeak) [reply tests/gold_tests/pluginTest/money_trace/money_trace.test.py]: removed

File: tests/gold_tests/pluginTest/money_trace/money_trace.test.py (hunk 2 above)
Comment (traeak) [reply tests/gold_tests/pluginTest/money_trace/money_trace.test.py]: removed

**Review by randall — APPROVED** [review]

**Pull request merged**